contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,170 @@
+        individualBufferCap[_minter] = _bufferCap;
+        _bufferStored[_minter] = _bufferCap;
+
+        emit IndividualRateLimitPerSecondUpdate(_minter, 0, _rateLimitPerSecond);
+    }
+
+    /// @notice add an authorized minter contract
+    /// @param _minter the address whose buffer and rate limit per second will be set
+    /// @param _rateLimitPerSecond the new rate limit per second for this minter
+    /// @param _bufferCap  the new buffer cap for this minter
+    function updateAddress(address _minter, uint256 _rateLimitPerSecond, uint256 _bufferCap) external virtual override onlyGovernorOrAdmin {
+        require(_bufferCap <= bufferCap, "MultiRateLimited: buffercap too high");
+
+        individualRateLimitPerSecond[_minter] = _rateLimitPerSecond;
+        individualLastBufferUsedTime[_minter] = block.timestamp;
+        individualBufferCap[_minter] = _bufferCap;
+        _bufferStored[_minter] = _bufferCap;
+
+        emit IndividualRateLimitPerSecondUpdate(_minter, 0, _rateLimitPerSecond);
+    }
+
+    /// @notice remove an authorized minter contract
+    /// @param _minter the address to remove from the whitelist of addresses
+    function removeAddress(address _minter) external virtual override isGovernorOrGuardianOrAdmin {
+        uint256 oldRateLimitPerSecond = individualRateLimitPerSecond[_minter];
+
+        delete individualRateLimitPerSecond[_minter];
+        delete _bufferStored[_minter];
+        delete individualBufferCap[_minter];
+        individualLastBufferUsedTime[_minter] = block.timestamp;
+
+        emit IndividualRateLimitPerSecondUpdate(_minter, oldRateLimitPerSecond, 0);
+    }
+
+    /// @notice set the rate limit per second
+    /// @param minter the address whose buffer will be set
+    /// @param newRateLimitPerSecond the new rate limit per second for this minter
+    function setIndividualRateLimitPerSecond(address minter, uint256 newRateLimitPerSecond) external virtual override onlyGovernorOrAdmin {
+        require(newRateLimitPerSecond <= rateLimitPerSecond, "MultiRateLimited: rateLimitPerSecond too high");
+
+        _updateIndividualBufferStored(minter);
+        _setIndividualRateLimitPerSecond(minter, newRateLimitPerSecond);
+    }
+
+    /// @notice set the buffer cap
+    /// @param minter the address whose buffer will be set
+    /// @param newBufferCap the new buffer cap for this minter
+    function setIndividualBufferCap(address minter, uint256 newBufferCap) external virtual override onlyGovernorOrAdmin {
+        require(newBufferCap <= bufferCap, "MultiRateLimited: new buffer cap is over global max");
+
+        _setIndividualBufferCap(minter, newBufferCap);
+    }
+
+    /// @notice the amount of action used before hitting limit
+    /// @dev replenishes at rateLimitPerSecond per second up to bufferCap
+    /// @param minter the address whose buffer will be returned
+    function buffer(address minter) public view override returns(uint256) { 
```

Comment:
maybe name this `individualBuffer(address)` because all other methods are individualSomething. Also `buffer()` will return the global buffer, so it's confusing to have 2 functions with the same name

Reply:
good callout